src/webui/webapplication.cpp

```diff
@@ -173,45 +98,94 @@ namespace
+    {
+        const QRegExp regex("QBT_TR\\((([^\\)]|\\)(?!QBT_TR))+)\\)QBT_TR(\\[CONTEXT=([a-zA-Z_][a-zA-Z0-9_]*)\\])");
+        const QRegExp mnemonic("\\(?&([a-zA-Z]?\\))?");
+        int i = 0;
+        bool found = true;
+
+        const QString locale = Preferences::instance()->getLocale();
+        bool isTranslationNeeded = !locale.startsWith("en") || locale.startsWith("en_AU") || locale.startsWith("en_GB");
+
+        while (i < data.size() && found) {
+            i = regex.indexIn(data, i);
+            if (i >= 0) {
+                //qDebug("Found translatable string: %s", regex.cap(1).toUtf8().data());
+                QByteArray word = regex.cap(1).toUtf8();
+
+                QString translation = word;
+                if (isTranslationNeeded) {
+                    QString context = regex.cap(4);
+                    translation = qApp->translate(context.toUtf8().constData(), word.constData(), 0, 1);
+                }
+                // Remove keyboard shortcuts
+                translation.replace(mnemonic, "");
+
+                // Use HTML code for quotes to prevent issues with JS
+                translation.replace("'", "&#39;");
+                translation.replace("\"", "&#34;");
+
+                data.replace(i, regex.matchedLength(), translation);
+                i += translation.length();
+            }
+            else {
+                found = false; // no more translatable strings
+            }
+        }
+    }
 }
 
-void WebApplication::action_public_index()
+WebApplication::WebApplication(QObject *parent)
+    : AbstractWebApplication(parent)
+    , m_isAltUIUsed(false)
 {
-    QString path;
-
-    if (!args_.isEmpty()) {
-        if (args_.back() == "favicon.ico")
-            path = ":/icons/skin/qbittorrent16.png";
-        else
-            path = WWW_FOLDER + args_.join("/");
+    for (int i = 0; i < metaObject()->methodCount(); ++i) {
+        QMetaMethod method = metaObject()->method(i);
+        if (QLatin1String(method.tag()) == QLatin1String("PUBLIC_API"))
+            m_publicAPIs << method.name();
     }
 
-    printFile(path);
+    configure();
+    connect(Preferences::instance(), &Preferences::changed, this, &WebApplication::configure);
 }
 
-void WebApplication::action_public_webui()
+void WebApplication::sendWebUIFile()
 {
-    if (!sessionActive())
-        printFile(PRIVATE_FOLDER + "login.html");
-    else
-        printFile(PRIVATE_FOLDER + "index.html");
+    if (request().path == QLatin1String("/")) {
+        printFile(m_rootFolder
+                  + (sessionActive() ? PRIVATE_FOLDER : PUBLIC_FOLDER)
+                  + (sessionActive() ? QLatin1String("/index.html") : QLatin1String("/login.html")));
+    }
+    else if ((!m_isAltUIUsed) && (request().path.startsWith(PATH_PREFIX_IMAGES))) {
+        printFile(QLatin1String(":/icons/") + request().path.mid(PATH_PREFIX_IMAGES.size()));
+        header(Http::HEADER_CACHE_CONTROL, MAX_AGE_MONTH);
+    }
+    else {
```

Comment:
About the `if... else if ... else`, I would use 3 `if ()`, its clearer to me this way.

Reply:
Code reformatted.